internal/jimm/juju/jimm.go

```diff
@@ -328,3 +329,53 @@ func (j *JujuManager) PrepareModelMigration(
 
 	return migrationToken, nil
 }
+
+// ListMigrationTargets returns the list of juju controllers that the given internal
+// model could be migrated to. This includes controllers that support the model's
+// cloud region and version, but excludes the controller the model is already on.
+func (j *JujuManager) ListMigrationTargets(ctx context.Context, user *openfga.User, modelTag names.ModelTag) ([]dbmodel.Controller, error) {
+	const op = errors.Op("jimm.ListMigrationTargets")
+
+	if !user.JimmAdmin {
+		return nil, errors.E(op, errors.CodeUnauthorized, "unauthorized")
+	}
+
+	var model dbmodel.Model
+	model.SetTag(modelTag)
+	if err := j.Database.GetModel(ctx, &model); err != nil {
+		return nil, errors.E(op, err)
+	}
+
+	currentVersion, err := version.Parse(model.Controller.AgentVersion)
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+
+	var controllers []dbmodel.Controller
+	err = j.Database.ForEachController(ctx, func(c *dbmodel.Controller) error {
```

Comment:
i wonder if it would be slightly more efficient to look at the current model, get it's cloud region (FindRegionByCloudName), then loop through the list of controllers for that cloud region.